apps/agreement/contracts/Agreement.sol

```diff
@@ -648,19 +644,30 @@ contract Agreement is ILockManager, IAgreement, AragonApp {
     }
 
     /**
-    * @dev Tell the amount of leftover arbitration fees that the submitter must pay in order to to raise a dispute for the given action
-    * @param _actionId Identification number of the action being queried
-    * @return feeToken ERC20 token to be used for the arbitration fees
-    * @return missingFees Amount of arbitration fees missing to be able to dispute the action
-    * @return totalFees Total amount of arbitration fees required by the arbitrator to raise a dispute
+    * @dev Tell the fees paid for an action challenge
+    *      Split from getChallenge due to “stack too deep issues”
+    * @param _challengeId Identification number of the challenge being queried
+    * @return challengerArbitratorFeeAmount Amount of arbitration fees paid by the challenger in advance in case the challenge is raised to the arbitrator
+    * @return challengerArbitratorFeeToken ERC20 token used for the arbitration fees paid by the challenger in advance
+    * @return submitterArbitratorFeeAmount Amount of arbitration fees paid by the submitter when the challenge is raised to the arbitrator
+    * @return submitterArbitratorFeeToken ERC20 token used for the arbitration fees paid by the submitter
     */
-    function getMissingArbitratorFees(uint256 _actionId) external view returns (ERC20 feeToken, uint256 missingFees, uint256 totalFees) {
-        (Action storage action, Challenge storage challenge, ) = _getChallengedAction(_actionId);
-        IArbitrator arbitrator = _getArbitratorFor(action);
-        ERC20 challengerFeeToken = challenge.arbitratorFeeToken;
-        uint256 challengerFeeAmount = challenge.arbitratorFeeAmount;
+    function getChallengeFees(uint256 _challengeId)
+        external
+        view
+        returns (
+            uint256 challengerArbitratorFeeAmount,
+            ERC20 challengerArbitratorFeeToken,
+            uint256 submitterArbitratorFeeAmount,
+            ERC20 submitterArbitratorFeeToken
```

Comment:
Is this needed due to a "stack too deep" issue in `getChallenge` or was it due to sth else?

Reply:
Yes, I put a comment a few lines above. I’m going to try your solution of the structs.

Reply:
EDIT: ops, nope, I would need ABI v2.